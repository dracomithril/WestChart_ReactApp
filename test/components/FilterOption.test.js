/**
 * Created by XKTR67 on 4/19/2017.
 */
import React from "react";
import PropTypes from "prop-types";
import FilteringOption from "../../src/components/FilteringOptions/FilterOption";
import {shallow} from "enzyme";
import {shallowToJson} from "enzyme-to-json";
import configureMockStore from "redux-mock-store";
const initial_state = require('../data/initial_state.json');
const mockStore = configureMockStore([]);
describe('<FilteringOption/>', () => {
        let elem = {
            control: {name: "date_create_control", id: 'create'},
            input: {max: 31, name: "create_control"},
            valueName: "created_time",
            description: {start: 'created in last ', end: ' days'},
            type: 'countDays'
        };
    it('renders without crashing ChartPresenter', () => {
        const store = mockStore(initial_state);
        const wrapper = shallow(
            <FilteringOption {...elem} />, {
                context: {store},
                childContextTypes: {store: PropTypes.object}
            }
        );
        expect(shallowToJson(wrapper)).toMatchSnapshot();

    });
    it('should be able to change state of woc checkbox', () => {
        const store = mockStore(initial_state);
        const wrapper = shallow(
            <FilteringOption {...elem} />, {
                context: {store},
                childContextTypes: {store: PropTypes.object}
            }
        );
        expect(shallowToJson(wrapper)).toMatchSnapshot();
        const wocCheckbox = wrapper.findWhere(n => n.props().id === 'create').first();
        expect(wocCheckbox.props().checked).toBeTruthy();
        wocCheckbox.simulate('change',{target:{id:'create',checked:false}});
        let actions = store.getActions();
        expect(actions[0].type).toBe('TOGGLE_FILTER');
        expect(actions[0].id).toBe('create');
        expect(actions[0].checked).toBeFalsy();

    })

});
