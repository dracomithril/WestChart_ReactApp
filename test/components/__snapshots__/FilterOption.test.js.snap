// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilteringSettings/> renders without crashing ChartPresenter 1`] = `
<Checkbox
  bsClass="checkbox"
  checked={true}
  disabled={false}
  id="create"
  inline={false}
  name="date_create_control"
  onChange={[Function]}
>
  created in last 
  <input
    className="num_days"
    id="create"
    min={0}
    onChange={[Function]}
    step={1}
    type="number"
  />
   days
</Checkbox>
`;
